x-pack/plugins/reporting/server/browsers/chromium/driver/chromium_driver.ts

```diff
@@ -84,8 +88,29 @@ export class HeadlessChromiumDriver {
     return result;
   }
 
-  public waitForSelector(selector: string) {
-    return this.page.waitFor(selector);
+  public async waitForSelector(selector: string, opts: WaitForSelectorOpts = {}) {
+    const { silent = false } = opts;
+    this.logger.debug(`waitForSelector ${selector}`);
+
+    let resp;
+    try {
+      resp = await this.page.waitFor(selector);
+    } catch (err) {
+      if (!silent) {
+        // Provide some troubleshooting info to see if we're on the login page,
+        // "Kibana could not load correctly", etc
+        this.logger.error(`waitForSelector ${selector} failed on ${this.page.url()}`);
+        const pageText = await this.evaluate({
+          fn: () => document.querySelector('body')!.innerText,
```

Comment:
You probably don't want the bang here: 

`fn: () => document.querySelector('body').innerText,`

Reply:
The `!` is the "[Non-null assertion operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html)" for Typescript to not fail compilation for the reason that the return from `document.querySelector('body')` is possibly null. It's not as safe as a guard, though. It's possible for `querySelector` to return `null` if an element isn't found, but for `body` I don't see it ever happening.

Reply:
TIL! Does the TS compiler remove this when built? Reason I ask is because this function gets stringified and sent over WebSockets over to the browser... so if it doesn't it'll have issues running